### ISSUE TITLE
Pymoo problem & population definitions

### ISSUE THREAD

**Issue by BenAbyss** [anyoptimization/pymoo#148]
Hi, terribly sorry if this is the wrong place for forum questions.
I'm trying to create an optimisation system that uses a custom variable(using the provided guide on the website) for my dissertation. This variable in question has several weighted tags it wants to optimise. I'm having a consistent exception of 'Number of values and population size do not match!', and one of 'Input dimension 20 are not equal to n_var 1!' if i attempt to use any population size bar 1. My current Problem program code is:

    def __init__(self, tag_num):
        self.n_obj = tag_num
        super().__init__(n_var=1, n_obj=tag_num, n_constr=0, elementwise_evaluation=True)

    def _evaluate(self, x, out, *args, **kwargs):
        fitness_values = x[0].get_tag_fitness_values()
        for i in range(self.n_obj):
            try:
                fitness_values[i] *= -1
            except IndexError:
                fitness_values[i] = 0
        out["F"] = np.column_stack(fitness_values)

The method get_tag_fitness_value() successfully pulls an array of floats with a size equal to tag_num

**Comment by julesy89** [comment]
If you enable `elementwise_evaluation=True` then the value you set to `out["F"]` is supposed to be a vector (NumPy array with a shape of length one) of length `n_obj`.
Debugging shall show you that this is not the case right now.

**Comment by BenAbyss** [comment]
I did have it as that originally, using the code:
        out["F"] = np.array(fitness_values, dtype=float)
However, this still gave the same error back.

**Comment by julesy89** [comment]
check out the shape `out["F"]` after setting it.

**Comment by julesy89** [comment]
Can you give me the full example code that I can able to run it?

**Comment by BenAbyss** [comment]
Due to the nature of the project, I'm unsure/wary of how much sample code I can send, mainly due to a lot of it being interlinked with a database, so can't get it to a runnable state. I've included a zip file that has all the Pymoo classes I've attempted to overwrite, a sample file of how I'm calling the NSGA2 system. There's also a very basic diagram of the custom objects contained within the customised object I'm using the system for - they aren't interacted with directly but I thought I would include it just in case.

[SampleCode.zip](https://github.com/msu-coinlab/pymoo/files/6244203/SampleCode.zip)
